onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -1297,6 +1299,7 @@ Status MIGraphXExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>&
       if (!input_shape_match) {
         if (!load_precompiled_model(prog, load_compiled_model_, std::string{load_compiled_path_})) {
           LOGS_DEFAULT(VERBOSE) << "No Input shapes mismatch detected. Recompiling" << std::endl;
+          cmp_options.set_external_data_path(model_path_.parent_path().string());
```

Comment:
Could you do some test model paths like "model.onnx" and "./model.onnx" ? Sometime, the path may not have parent path. If that is so, you might need something like `model_path_.has_parent_path() ? model_path_.parent_path() : std::filesystem::current_path()`

Reply:
Sure added